test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -164,12 +165,30 @@ UcxConnection* UcxContext::connect(const struct sockaddr* saddr, size_t addrlen)
 
 void UcxContext::progress()
 {
-    ucp_worker_progress(_worker);
-    progress_io_message();
+    for (unsigned i = 0; i < 1000; ++i) {
+        bool progressed = ucp_worker_progress(_worker);
+        progress_io_message();
+        if (!progressed) {
+            break;
+        }
+    }
+
     progress_conn_requests();
     progress_failed_connections();
 }
 
+size_t UcxContext::n_connections() const {
+    return _conns.size();
+}
+
+size_t UcxContext::n_conn_requested() const {
+    return _n_conn_requested;
+}
+
+size_t UcxContext::n_conn_accepted() const {
+    return _n_conn_accepted;
+}
```

Comment:
can we do this only in io_demo.cc w/o changing ucx_wrapper.cc (e.g by virtual function override on connect/disconnect)

Reply:
do you mean adding new functions to wrapper? since all connection requests and connections of server itself are handled in wrapper.

Reply:
done